### PR TITLE
groff: local config should survive upgrade

### PR DESCRIPTION
**Pull request by sshambar**
Upstream groff places "site-local" files under @datadir@, but in Homebrew @datadir@ is overwritten on upgrades.  Placing site-local files under #{etc} allows local customizations to be retained on upgrade.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/g/groff.rb [hunk 1]
```diff
@@ -36,7 +37,13 @@ class Groff < Formula
...
 
   def install
-    system "./configure", "--prefix=#{prefix}", "--without-x", "--with-uchardet"
+    # Local config needs to survive upgrades
+    inreplace "Makefile.in", "localfontdir = $(dataprogramdir)",
+              "localfontdir = @sysconfdir@/groff"
+    inreplace "Makefile.in", "localtmacdir = $(dataprogramdir)",
+              "localtmacdir = @sysconfdir@/groff"
+    system "./configure", "--prefix=#{prefix}", "--sysconfdir=#{etc}",
+           "--without-x", "--with-uchardet"
```
Comment (SMillerDev) [inline Formula/g/groff.rb:46]:
```suggestion
    system "./configure", "--sysconfdir=#{etc}", "--without-x", "--with-uchardet", *std_configure_args
```

Comment (sshambar) [reply Formula/g/groff.rb:46]: @SMillerDev : I tried using the std_configure_args earlier, but one of the standard args isn't recognized...

File: Formula/g/groff.rb [hunk 2]
```diff
@@ -36,7 +37,13 @@ class Groff < Formula
   end
 
   def install
-    system "./configure", "--prefix=#{prefix}", "--without-x", "--with-uchardet"
+    # Local config needs to survive upgrades
+    inreplace "Makefile.in", "localfontdir = $(dataprogramdir)",
+              "localfontdir = @sysconfdir@/groff"
+    inreplace "Makefile.in", "localtmacdir = $(dataprogramdir)",
+              "localtmacdir = @sysconfdir@/groff"
```
Comment (carlocab) [inline Formula/g/groff.rb:44]:
```suggestion
    inreplace "Makefile.in" do |s|
      s.change_make_var! "localfontdir", "@sysconfdir@/groff"
      s.change_make_var! "localtmacdir", "@sysconfdir@/groff"
    end
```

File: Formula/g/groff.rb (hunk 1 above)
Comment (sshambar) [reply Formula/g/groff.rb:46]: @SMillerDev: actually --disable-debug, although not recognized, only issues a WARNING -- I'll add it.

File: Formula/g/groff.rb (hunk 2 above)
Comment (sshambar) [reply Formula/g/groff.rb:44]: @carlocab - nice feature, I'll add it (although there's more to the values, I'll edit...)

**Comment by sshambar** [comment]
@carlocab - there's a series of patches in debian -- should I bother to add some of those, or just leave this PR focused on the config?

**Comment by carlocab** [comment]
Let's skip the patches for now.

**Review by carlocab — APPROVED** [review]
Thanks! Apologies for the delay here.

**Pull request merged**